ipn/ipnlocal/local.go

```diff
@@ -666,11 +666,19 @@ func (b *LocalBackend) linkChange(delta *netmon.ChangeDelta) {
 	}
 }
 
-func (b *LocalBackend) onHealthChange(sys health.Subsystem, err error) {
-	if err == nil {
-		b.logf("health(%q): ok", sys)
+func (b *LocalBackend) onHealthChange(w *health.Warnable, us *health.UnhealthyState) {
+	if us == nil {
+		b.logf("health(warnable=%s): ok", w.Code)
 	} else {
-		b.logf("health(%q): error: %v", sys, err)
+		b.logf("health(warnable=%s): error: %s", w.Code, us.Text)
+	}
+
+	// Notify the frontend of the current health state if available.
+	state := b.health.CurrentState()
+	if state != nil {
+		b.send(ipn.Notify{
+			Health: state,
+		})
 	}
```

Comment:
this never sends an update when health warnings clear back to all happy. In that case CurrentState is nil, but if the previous one was non-nil, then you need to send a ipn.Notify with a non-nil health.State (to signal a change) with an empty-or-nil map inside.

Reply:
Done. We now always provide a *State, but the Warnings map within might be nil (to indicate a fully healthy state).